R/rd-data.R

```diff
@@ -40,6 +40,52 @@ as_data.tag_item <- function(x, ...) {
 parse_section <- function(x, title, ...) {
   text <- flatten_para(x, ...)
 
+  html <- xml2::read_html(text)
+  r_blocks <- xml2::xml_find_all(html, "//div[contains(@class, 'sourceCode r')]/pre/code")
+
+  highlight_r_block <- function(block) {
+    xml2::xml_text(block) <- downlit::highlight(
+      xml_text1(block),
+      classes = downlit::classes_pandoc()
+    )
+  }
+
+  purrr::walk(r_blocks, highlight_r_block)
+
+  non_r_blocks <- xml2::xml_find_all(
+    html,
+    "//div[contains(@class, 'sourceCode') and not(contains(@class, 'sourceCode r'))]"
```

Comment:
the XPath query is different because we need to keep the div until we extract the language.